org.eclipse.scanning.points/src/org/eclipse/scanning/points/serialization/PositionDeserializer.java

```diff
@@ -25,10 +33,57 @@
 
 	@Override
 	public IPosition deserialize(JsonParser parser, DeserializationContext context) throws IOException, JsonProcessingException {
+		PositionBean<?> bean = parser.readValueAs(PositionBean.class);
+		return fromBean(bean);
+	}
+
+	private <T extends IPosition> IPosition fromBean(PositionBean<T> bean) throws IOException {
+		final Map<String, Object> values = bean.getValues();
+		final Map<String, Integer> indices = bean.getIndices();
+		final int stepIndex = bean.getStepIndex();
+		final List<Collection<String>> dimensionNames = bean.getDimensionNames();
+		final Class<?> klass = bean.getPositionClass();
+
+		// TODO: a private method for each type, or move to factory method on position class, e.g. MapPosition.fromPositionBean()
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)